homeassistant/components/amcrest/__init__.py

```diff
@@ -157,7 +179,7 @@ def setup(hass, config):
                     CONF_SWITCHES: switches
                 }, config)
 
-    return True
+    return len(hass.data[DATA_AMCREST]) >= 1
```

Comment:
In which case it would return False?

Reply:
It would return False if none of the configured cameras worked. If you look at the top of the loop, it tries to communicate with each configured camera. If that fails (AmcrestError), then the camera is not appended to the list. If none work, the list will be empty. In that case setup will return False. Is that ok?